sonar-python-plugin/src/main/java/org/sonar/plugins/python/TextReportReader.java

```diff
@@ -31,11 +31,19 @@
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
 
-public class Flake8ReportReader {
+/**
+ * Common implementation to parse Flake8 and Pylint reports
+ */
+public class TextReportReader {
+
+  private final int reportOffset;
+  protected static final Pattern DEFAULT_PATTERN = Pattern.compile("(.+):(\\d+):(\\d+): (\\S+[^:]):? (.*)");
+  protected static final Pattern LEGACY_PATTERN = Pattern.compile("(.+):(\\d+): \\[(.*)\\] (.*)");
+  private static final Logger LOG = Loggers.get(TextReportReader.class);
 
-  private static final Logger LOG = Loggers.get(Flake8ReportReader.class);
-  private static final Pattern DEFAULT_PATTERN = Pattern.compile("(.+):(\\d+):(\\d+): (\\S+) (.*)");
-  private static final Pattern PYLINT_PATTERN = Pattern.compile("(.+):(\\d+): \\[(.*)\\] (.*)");
+  public TextReportReader(int columnStartIndex) {
```

Comment:
_minor, subjective_ Having to call this constructor with some magic numbers seems a little off to me. I think it would look nicer if we introduced integer constants and used them something like this:
```
new TextReportReader(TextReportReader.COLUMN_ZERO_BASED)
new TextReportReader(TextReportReader.COLUMN_ONE_BASED)
```

Feel free to disregard if you feel that this is too much.